src/topology/vector_bundle.lean

```diff
@@ -503,3 +536,279 @@ lemma to_topological_vector_bundle :
+  refine continuous_on.prod' _ _,
+  { refine ((continuous_snd.comp_continuous_on ψ₁.continuous_on).comp
+      (continuous_id.prod_map continuous_fst).continuous_on _).congr _,
+    { rw hψ₁,
+      mfld_set_tac },
+    { rintros ⟨x, ⟨w₁, w₂⟩⟩ ⟨hx, -⟩,
+      have hxe₁ : x ∈ e₁.base_set := hx.1.1,
+      have hxe₂ : x ∈ e₂.base_set := hx.1.2,
+      dsimp,
+      rw [prod_symm_apply hx.2.1 hx.2.2, prod_apply hxe₁ hxe₂],
+      dsimp,
+      rw [f₁.symm_apply_eq_prod_continuous_linear_equiv_at_symm] } },
+  { refine ((continuous_snd.comp_continuous_on ψ₂.continuous_on).comp
+      (continuous_id.prod_map continuous_snd).continuous_on _).congr _,
+    { rw hψ₂,
+      mfld_set_tac },
+    { rintros ⟨x, ⟨w₁, w₂⟩⟩ ⟨hx, -⟩,
+      have hxe₁ : x ∈ e₁.base_set := hx.1.1,
+      have hxe₂ : x ∈ e₂.base_set := hx.1.2,
+      dsimp,
+      rw [prod_symm_apply hx.2.1 hx.2.2, prod_apply hxe₁ hxe₂],
+      dsimp,
+      rw [f₂.symm_apply_eq_prod_continuous_linear_equiv_at_symm] } },
+end
+
+end pretrivialization
+
+open pretrivialization
+
+variables [topological_vector_bundle R F₁ E₁] [topological_vector_bundle R F₂ E₂]
+
+/-- The direct sum of topological vector bundles is a `topological_vector_prebundle` (this is an
+auxiliary construction for the `topological_vector_prebundle` instance, in which the
+pretrivializations are collated but no topology on the total space is yet provided). -/
+def _root_.vector_bundle_prod.topological_vector_prebundle :
+  topological_vector_prebundle R (F₁ × F₂) (vector_bundle_prod R F₁ E₁ F₂ E₂) :=
+{ pretrivialization_at := λ x,
+    pretrivialization.prod (trivialization_at R F₁ E₁ x) (trivialization_at R F₂ E₂ x),
+  mem_base_pretrivialization_at := λ x,
+    ⟨mem_base_set_trivialization_at R F₁ E₁ x, mem_base_set_trivialization_at R F₂ E₂ x⟩,
+  continuous_triv_change := λ p q, pretrivialization.continuous_triv_change_prod
+    (trivialization_at R F₁ E₁ p)
+    (trivialization_at R F₁ E₁ q)
+    (trivialization_at R F₂ E₂ p)
+    (trivialization_at R F₂ E₂ q),
+  total_space_mk_inducing := λ b,
+  begin
+    let e₁ := trivialization_at R F₁ E₁ b,
+    let e₂ := trivialization_at R F₂ E₂ b,
+    have hb₁ : b ∈ e₁.base_set := mem_base_set_trivialization_at R F₁ E₁ b,
+    have hb₂ : b ∈ e₂.base_set := mem_base_set_trivialization_at R F₂ E₂ b,
+    have key : inducing (λ w : E₁ b × E₂ b,
+      (b, e₁.continuous_linear_equiv_at b hb₁ w.1, e₂.continuous_linear_equiv_at b hb₂ w.2)),
+    { refine (inducing_prod_mk b).comp _,
+      exact ((e₁.continuous_linear_equiv_at b hb₁).to_homeomorph.inducing.prod_mk
+        (e₂.continuous_linear_equiv_at b hb₂).to_homeomorph.inducing) },
+    { convert key,
+      ext1 w,
+      simpa using prod_apply hb₁ hb₂ w.1 w.2 },
+  end }
+
+/-- The natural topology on the total space of the product of two vector bundles. -/
+instance : topological_space (total_space (vector_bundle_prod R F₁ E₁ F₂ E₂)) :=
+(vector_bundle_prod.topological_vector_prebundle R F₁ E₁ F₂ E₂).total_space_topology
```

Comment:
This topology is a complicated object, and we will never really use its definition. I wonder if it would be worth making the definition irreducible to help Lean (i.e., make an irreducible definition, and then register the instance using this irreducible definition). It would probably just amount to using a few `rw` in the next proofs, to unfold the definition, and then once the API is here it should never be necessary any more.